wfe2/wfe.go

```diff
@@ -829,21 +829,55 @@ func (wfe *WebFrontEndImpl) revokeCertByKeyID(
 	// certificate by checking that the account has valid authorizations for all
 	// of the names in the certificate or was the issuing account
 	authorizedToRevoke := func(parsedCertificate *x509.Certificate) *probs.ProblemDetails {
-		cert, err := wfe.SA.GetCertificate(ctx, core.SerialToString(parsedCertificate.SerialNumber))
-		if err != nil {
+		// Try to find a stored final certificate for the serial number
+		serial := core.SerialToString(parsedCertificate.SerialNumber)
+		cert, err := wfe.SA.GetCertificate(ctx, serial)
+		if berrors.Is(err, berrors.NotFound) && features.Enabled(features.PrecertificateRevocation) {
+			// If there was an error, it was a not found error, and the precertificate
+			// revocation feature is enabled, then try to find a stored precert.
+			pbCert, err := wfe.SA.GetPrecertificate(ctx,
+				&sapb.Serial{Serial: &serial})
+			if berrors.Is(err, berrors.NotFound) {
+				// If looking up a precert also returned a not found error then return
+				// a not found problem.
+				return probs.NotFound("No such certificate")
+			} else if err != nil {
+				fmt.Printf("\n\nERR: %#v\n\n", err)
```

Comment:
Leftover debug.

Reply:
D'oh! good catch, thanks. Fixed in 4e159e3